sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,102 @@
+    private static final String EQUAL = "=";
+    private static final String SEMICOLON = ";";
+    private static final String SEQUENCE_NUMBER_CANNOT_PARSED = "Sequence number cannot be parsed to long.";
+    private static final String VALID_FORMAT_ERROR_MESSAGE =
+        "Expected sync-token valid format should be <id>=<value>;sn=<sn>. For multiple sync tokens, " +
+            "<id>=<value>;sn=<sn>,<id>=<value>;sn=<sn>.";
+
+    private final ClientLogger logger = new ClientLogger(SyncToken.class);
+
+    private final String id;
+    private final String value;
+    private final Long sequenceNumber;
+
+    /**
+     * Create an instance of SyncToken class. Skip the sync token if the parsing failed.
+     *
+     * @param syncToken only one raw sync-token string from HTTP response header, ex.,
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn></p>}. But not
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn>,<id>=<value>;sn=<sn>}</p>
+     * Allows for better concurrency and client cache ability. The client may choose to use only token's last version,
+     * since token versions are inclusive. Not required for requests.
+     */
+    SyncToken(String syncToken) {
+        if (CoreUtils.isNullOrEmpty(syncToken)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(CANNOT_EMPTY_OR_NULL));
+        }
+
+        final String[] syncTokenParts = syncToken.split(SEMICOLON, 2);
+        if (syncTokenParts.length != 2) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(VALID_FORMAT_ERROR_MESSAGE));
+        }
+
+        final String[] idParts = syncTokenParts[0].split(EQUAL, 2);
+        final String[] snParts = syncTokenParts[1].split(EQUAL, 2);
+        if (idParts.length != 2 || snParts.length != 2
+                || idParts[0].isEmpty() || idParts[1].isEmpty()
+                || snParts[0].isEmpty() || snParts[1].isEmpty()) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(VALID_FORMAT_ERROR_MESSAGE));
+        }
+
+        try {
+            this.sequenceNumber = Long.parseLong(snParts[1]);
+        } catch (NumberFormatException ex) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(SEQUENCE_NUMBER_CANNOT_PARSED));
+        }
+        this.id = idParts[0];
+        this.value = idParts[1];
+    }
+
+    /**
+     * Get Token ID (opaque).
+     *
+     * @return Token ID
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Get Token value (opaque). Allows base64 encoded string.
+     *
+     * @return Token value
+     */
+    public String getValue() {
+        return value;
+    }
+
+    /**
+     * Get Token sequence number (version).
+     *
+     * Higher means newer version of the same token. Allows for better concurrency
+     * and client cache ability. The client may choose to use only token's last version, since token versions are
+     * inclusive. Not required for requests.
```

Comment:
They may not be required for requests but can they be used in requests?

Reply:
Sequence number will not be used in the request.